node/src/types/block.rs

```diff
@@ -1000,8 +1000,11 @@ impl BlockSignatures {
         self.proofs.insert(public_key, signature)
     }
 
-    pub(crate) fn has_proof(&self, public_key: &PublicKey) -> bool {
-        self.proofs.contains_key(public_key)
+    pub(crate) fn has_proof(&self, public_key: &PublicKey, signature: &Signature) -> bool {
+        self.proofs
+            .get(public_key)
+            .iter()
+            .any(|sig| *sig == signature)
```

Comment:
(Shouldn't be necessary; see other comments.)

Reply:
Done in [`f5f6038` (#1268)](https://github.com/CasperLabs/casper-node/pull/1268/commits/f5f603830d33577186c4c7072cbb578714d8e773)